### ISSUE TITLE
Support for CPU-only Inference/Evaluation (No GPU) – Shape Mismatch Error

### ISSUE THREAD

**Issue by abhinav3632** [OpenDriveLab/UniAD#261]
Hi,
I am trying to run inference/evaluation (not training) using the pre-trained UniAD models on a system with only an integrated Intel GPU (no dedicated NVIDIA GPU). I understand there will be a performance drop, but I only want to test the evaluation pipeline and outputs.

However, when running the evaluation script, I encounter the following error:
`Error: The size of tensor a (4) must match the size of tensor b (16) at non-singleton dimension 3
  ...
  File "mmcv/ops/multi_scale_deform_attn.py", line 153, in multi_scale_deformable_attn_pytorch
    output = (torch.stack(sampling_value_list, dim=-2).flatten(-2) *`

I believe this is related to the lack of CUDA support and the fallback to CPU ops.

Questions:

1. Is it possible to run UniAD inference/evaluation on CPU only, or is a CUDA-enabled GPU strictly required even for evaluation?
2. If CPU-only evaluation is not currently supported, are there any workarounds or plans to support this in the future?
3. If not supported, could you provide guidance on what would be required to enable CPU-only inference (even with reduced performance)?

Thank you for your help!
